lib/array_equals.rb

```diff
@@ -1,5 +1,18 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == nil && array2 == nil || array1 == [] && array2 == []
+    return true
+  elsif array1 == nil || array2 == nil || array1.length != array2.length
+    return false
+  else
+    i = 0
```

Comment:
Nit: this block shouldn't be nested inside an `else` (because the `else` is redundant here).